active/0000-pluggableLintTestGenerator.md

```diff
@@ -0,0 +1,61 @@
+- Start Date: 2015-05-19
+- RFC PR: (leave this empty)
+- ember-cli Issue: (leave this empty)
+
+# Summary
+
+Provide a unified API for linting for complete choice of testing framework and linting platform.
+
+# Motivation
+
+Allow user to have full control over what testing language and what linter they prefer.
+Currently ember-cli is tied into using JSHint always rather than being fully agnostic.
+
+# Detailed design
+
+In ember-cli addons main file.
+
+Expose a test generator API for the linting addon to call:
+```
+  lintTree: function(type, tree) {
+    return jshintTrees(tree, {
+      jshintrcPath: this.jshintrc.tests,
+      description: 'JSHint ' + type,
+      testGenerator: this.testGenerator
+    });
+  }
+```
+
+Expose a new test Generator hook that would return the lint errors transformed to the test language (This will likely be in the testing framework addon):
+```
+  testGenerator: function (relativePath, errors) {
```

Comment:
I was thinking the nicest would be defining an object shape for these which both JSHint and ESLint seem to do anyway.
Essentially:
[
  {
    line: 12,
    column: 30,
    ruleId: 1,
    message: 'error name'
  }
]

I was thinking it would be nice if the objects extended from `new Error` but I'm not really fussy so long as we have a defined API.